app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt

```diff
@@ -41,8 +45,7 @@ class CalendarActivity : AppCompatActivity() {
 
         // Add the new event to the deadline list
         val deadline = Deadline(deadlineTitle, DeadlineState.TODO, selectedDate)
-        val adapter = DeadlineAdapter(this, listOf(deadline))
-        listView.adapter = adapter
+        runBlocking { deadlineRepository.put(deadline) }
 
```

Comment:
why do you need a runBlocking op here?

Reply:
`deadlineRepository.put` is a suspend function (i.e. asynchronous). `runBlocking` will await the operation in the main thread, but the UI thread. This way we can wait for the `put` operation to complete before showing the result in the UI.